lib/internal/per_context/primordials.js

```diff
@@ -102,18 +115,16 @@ primordials.SafePromise = makeSafe(
   'Set',
   'String',
   'Symbol',
-].forEach((name) => {
-  const original = global[name];
-  const target = primordials[name] = Object.setPrototypeOf({
-    [name]: function(...args) {
-      return new.target ?
-        ReflectConstruct(original, args, new.target) :
-        ReflectApply(original, this, args);
-    }
-  }[name], null);
-  copyProps(original, target);
-  const proto = primordials[name + 'Prototype'] = Object.create(null);
-  copyPrototype(original.prototype, proto);
+].forEach((n) => {
+  copy(n, globalThis[n]);
+});
+
+// These don't have real constructors
+[
+  ['MapIteratorPrototype', Object.getPrototypeOf(new Map()[Symbol.iterator]())]
+].forEach(([n, v]) => {
+  primordials[n] = Object.create(null);
+  copyPrototype(v, primordials[n]);
 });
```

Comment:
Why are the changes in this file needed?

Reply:
[MapIteratorPrototype](https://tc39.es/ecma262/#sec-%mapiteratorprototype%-object) is not a global , it needs to be accessed via a prototype on a result of a method call. This makes it available on the primordials.